rmf_task/test/mock/MockDelivery.hpp

```diff
@@ -60,6 +60,18 @@ class MockDelivery : public rmf_task::Task
 
       // TODO(MXG): We could use the _restored_state here to fast-forward this
       // mock task to the phase that the previous Task::Active left off from.
+      if (_restored_state.has_value())
+      {
+        uint64_t phase_id = std::stoul(_restored_state.value());
+        const uint64_t min_phase_id = 0;
+        const uint64_t max_phase_id = pending_phases().size();
+        if (phase_id < min_phase_id && phase_id > max_phase_id) {
+          throw std::runtime_error("[MockDelivery::Active] phase_id given was " + std::to_string(phase_id) + " not in range of " + std::to_string(min_phase_id) + "," + std::to_string(max_phase_id) + ".");
+        }
+        for (int i = 0; i < phase_id; i++)
+          // BH: Is a spam of "start next phase" the right way to fast forward?
```

Comment:
No, this is not the way we should restore the task to its correct phase. This approach would give the impression that the robot is suddenly blasting through its phases, which is not what's actually supposed to happen. Instead the task should behave as though the other phases are already completed and not even worth mentioning.

The task should also be **in** the phase that it's being restored to by the time the task is returned from this function. It should not be necessary to call `start_next_phase` externally to enter the phase that's being returned to.

I think the simplest way to take care of the skipping would be to do
```
_pending_phases.erase(_pending_phases.begin(), _pending_phases.begin()+phase_id);
```
and then run `start_next_phase` just once before exiting this function.